config/v3_1_experimental/types/ignition.go

```diff
@@ -32,9 +35,31 @@ func (c ConfigReference) Key() string {
 
 func (cr ConfigReference) Validate(c path.ContextPath) (r report.Report) {
 	r.AddOnError(c.Append("source"), validateURLNilOK(cr.Source))
+	r.AddOnError(c.Append("httpHeaders"), cr.validateSchemeForHTTPHeaders())
 	return
 }
 
+func (cr ConfigReference) validateSchemeForHTTPHeaders() error {
+	if util.NilOrEmpty(cr.Source) {
+		return nil
+	}
+	u, err := url.Parse(*cr.Source)
+	if err != nil {
+		return errors.ErrInvalidUrl
+	}
+
+	if cr.HTTPHeaders == nil {
```

Comment:
Nit: this should probably be moved to the top of the function body.

Reply:
done